### PR TITLE
jclouds location customizer supports pre-release and post-release callback

### PR DESCRIPTION
**Pull request by ahgittin**
useful when the customizer is e.g. setting up then clearing external setup, e.g. DNS

### CODE REVIEW

**Comment by ggrunin** [comment]
+1 tested - it works!

File: locations/jclouds/src/main/java/brooklyn/location/jclouds/JcloudsLocation.java [hunk 1]
```diff
@@ -1764,24 +1769,57 @@ public void release(SshMachineLocation machine) {
...
 
         Exception tothrow = null;
+
+        if (machine instanceof JcloudsSshMachineLocation) {
+            ConfigBag setup = config().getBag();
+            for (JcloudsLocationCustomizer customizer : getCustomizers(setup)) {
+                try {
+                    customizer.preRelease((JcloudsSshMachineLocation) machine);
+                } catch (Exception e) {
+                    LOG.error("Problem invoking pre-release customizer "+customizer+" for machine "+machine+" in "+this+", instance id "+instanceId+
```
Comment (neykov) [inline locations/jclouds/src/main/java/brooklyn/location/jclouds/JcloudsLocation.java]: Worth adding `Exceptions.propagateIfFatal`

Comment (neykov) [inline locations/jclouds/src/main/java/brooklyn/location/jclouds/JcloudsLocation.java]: Or is the code trying to release aggressively and throwing here will lead to machine leaks...

Comment (ahgittin) [inline locations/jclouds/src/main/java/brooklyn/location/jclouds/JcloudsLocation.java]: the others were trying to release aggressively, throwing afterwards.  i've duplicated that pattern.

**Pull request merged**